neurodamus/metype.py

```diff
@@ -76,6 +159,7 @@ def __init__(self, gid, etype_path, emodel, morpho_path, meinfos=None, detailed_
         self.extra_attrs = None
 
         self._instantiate_cell(gid, etype_path, emodel, morpho_path, meinfos, detailed_axon)
+        self.set_section_counts()
```

Comment:
Currently, the attribute `self._section_counts` is required by functions `get_section_id` and `get_sec`. Can we call `set_section_counts` when these functions are called instead of all cases? For spikes and soma reports, we don't need them. 
If I don't misunderstand, @mgeplf mentioned similar ideas at the meeting.